src/figuresClasses.ts

```diff
@@ -1,19 +1,88 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
 }
 
 export class Triangle {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side should be positive');
+    }
+
+    if (a + b <= c
+      || a + c <= b
+      || c + b <= a) {
+      throw new Error('Your sides is not create triangle');
+    }
+  }
 
+  getArea(): number {
+    const halfPerimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(halfPerimeter
+      * (halfPerimeter - this.a)
+      * (halfPerimeter - this.b)
+      * (halfPerimeter - this.c)).toFixed(2);
+
+    return +area;
+  }
 }
 
 export class Circle {
+  public shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius should be positive');
+    }
+  }
+
+  getArea(): number {
+    const pi = +Math.PI.toFixed(2);
```

Comment:
In my opinion, the variable "pi" might be redundant here. You may use '3.14' instead since you're rounding up to hundreds.

Reply:
Agree, it is better to round the number just once in the end.